### PR TITLE
[Icons] Correct icons scripts and integration.

### PR DESCRIPTION
**Pull request by romoore**
Icons were not fully integrated into the catalog project and the relevant
scripts were out of date. This PR addressed the following issues:

*   The Icons source files had been hand-modified and did not match the
    script outputs.
*   The Icons unit tests were not included in the Catalog or Dragons
    projects.

It addresses these issues through the following:

*   Updates `scripts/sync_icons.sh` to:
    -   Match the copyright header format used by the project.
    -   Run clang-format on each generated source file.
    -   Add Icons unit tests to the MaterialComponents podspec.
    -   Remove duplicate import of `MaterialIcons.h` in the source
        implementation files.
*   Adds the Icons unit test targets to the Catalog and Dragons Xcode schemes.

Discovered while working on #8498

### CODE REVIEW

File: components/private/Icons/icons/ic_more_horiz/src/MaterialIcons+ic_more_horiz.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+// Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (andrewoverton) [inline components/private/Icons/icons/ic_more_horiz/src/MaterialIcons+ic_more_horiz.h:1]: Why 2016?

Comment (romoore) [reply components/private/Icons/icons/ic_more_horiz/src/MaterialIcons+ic_more_horiz.h:1]: I suspect that's when the script was written. I didn't modify that part in this PR.

File: scripts/sync_icons.sh [hunk 2]
```diff
@@ -200,35 +204,31 @@ EOL
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 // This file was automatically generated by running $0
 // Do not modify directly.
 
 #import "$file.h"
 
-#import "MaterialIcons.h"
```
Comment (andrewoverton) [inline scripts/sync_icons.sh:224]: I guess this wasn't necessary?

**Review by andrewoverton — APPROVED** [review]

File: scripts/sync_icons.sh (hunk 2 above)
Comment (romoore) [reply scripts/sync_icons.sh:224]: Nope, it's already imported in the headers.

**Pull request merged**